src/clustering/administration/main/command_line.cc

```diff
@@ -645,6 +649,163 @@ service_address_ports_t get_service_address_ports(const std::map<std::string, op
+
+    if (!(key_file && cert_file)) {
+        logERR("must specify both --web-tls-key and --web-tls-cert");
+        return false;
+    }
+
+    return load_tls_key_and_cert(web_tls, *key_file, *cert_file);
+}
+
+bool configure_driver_tls(const std::map<std::string, options::values_t> &opts, SSL_CTX *driver_tls) {
+    boost::optional<std::string> key_file = get_optional_option(opts, "--driver-tls-key");
+    boost::optional<std::string> cert_file = get_optional_option(opts, "--driver-tls-cert");
+    boost::optional<std::string> ca_file = get_optional_option(opts, "--driver-tls-ca");
+
+    if (!(key_file && cert_file)) {
+        logERR("must specify both --driver-tls-key and --driver-tls-cert");
+        return false;
+    }
+
+    if (!load_tls_key_and_cert(driver_tls, *key_file, *cert_file)) {
+        return false;
+    }
+
+    if (ca_file) {
+        if (!SSL_CTX_load_verify_locations(driver_tls, ca_file->c_str(), NULL)) {
+            ERR_print_errors_fp(stderr);
+            return false; 
+        }
+
+        // Mutual authentication.
+        SSL_CTX_set_verify(driver_tls, SSL_VERIFY_PEER|SSL_VERIFY_FAIL_IF_NO_PEER_CERT, NULL);
+    }
+
+    return true;
+}
+
+bool configure_cluster_tls(const std::map<std::string, options::values_t> &opts, SSL_CTX *cluster_tls) {
+    boost::optional<std::string> key_file = get_optional_option(opts, "--cluster-tls-key");
+    boost::optional<std::string> cert_file = get_optional_option(opts, "--cluster-tls-cert");
+    boost::optional<std::string> ca_file = get_optional_option(opts, "--cluster-tls-ca");
+
+    if (!(key_file && cert_file && ca_file)) {
+        logERR("must specify each of --cluster-tls-key, --cluster-tls-cert, and --cluster-tls-ca");
+        return false;
+    }
+
+    if (!load_tls_key_and_cert(cluster_tls, *key_file, *cert_file)) {
+        return false;
+    }
+
+    if (!SSL_CTX_load_verify_locations(cluster_tls, ca_file->c_str(), NULL)) {
+        ERR_print_errors_fp(stderr);
+        return false; 
+    }
+
+    // Mutual authentication.
+    SSL_CTX_set_verify(cluster_tls, SSL_VERIFY_PEER|SSL_VERIFY_FAIL_IF_NO_PEER_CERT, NULL);
+
+    return true;
+}
+
+bool initialize_tls_ctx(SSL_CTX **tls_ctx) {
+    *tls_ctx = SSL_CTX_new(SSLv23_method());
+    if (NULL == *tls_ctx) {
+        ERR_print_errors_fp(stderr);
+        return false;
+    }
+
+    // Only allow TLS v1.2 and prefer server ciphers.
+    SSL_CTX_set_options(*tls_ctx, SSL_OP_NO_SSLv2|SSL_OP_NO_SSLv3|SSL_OP_NO_TLSv1|SSL_OP_NO_TLSv1_1|SSL_OP_CIPHER_SERVER_PREFERENCE);
+
+    /* This is pretty important. We want to use the most secure TLS cipher
+    suite that we can. This simple list only allows ciphers suites which employ
+    ECDHE (Elliptic Curve Diffie-Hellman with Ephemeral keys) for encryption
+    key agreement for Perfect Forward Secrecy and we want to use AESGCM for
+    efficient authenticated encryption. */
+    if (0 == SSL_CTX_set_cipher_list(*tls_ctx, "ECDHE+AESGCM")) {
```

Comment:
Sure, let's have this be the default and I'll add another config/option for specifying an additional cipher list, maybe `--tls-additional-ciphers`.

Should we also have a `--tls-dh-params` flag too, for specifying diffie-hellman params for the non-EECDH ciphers? If not specified, we could just generate a 2048-bit dhparams file by default the first time that a node starts and save it in some default location. This is still a huge pain though because the server would be stalled for a very long time while it searches for a strong prime, so given that they already have to specify additional ciphers maybe it's not too much to ask for them to generate dhparams too?